src/nbl/video/COpenGLEvent.h

```diff
@@ -0,0 +1,22 @@
+#ifndef __NBL_C_OPENGL_EVENT_H_INCLUDED__
+#define __NBL_C_OPENGL_EVENT_H_INCLUDED__
+
+#include "nbl/video/IGPUEvent.h"
+#include "nbl/video/COpenGLSync.h"
+
+namespace nbl {
+namespace video
+{
+
+class COpenGLEvent : public IGPUEvent
+{
+public:
+    using IGPUEvent::IGPUEvent;
+
+private:
+    core::smart_refctd_ptr<COpenGLSync> m_syncSignal, m_syncUnsignal;
```

Comment:
Lets lock it down, only allow the user to create an event with the DEVICE_ONLY option

make all host operations (GetStatus and host reset) assert false and crash

the command buffer operations should just work out the `glMemoryBarrier` bits to use from the dependency infos specified during the set and wait method and carry out the barrier in the wait.

the commandbuffer reset should just clear the info saved from the last set operation.

the `COpenGLSync` can go unused for now.